### PR TITLE
Added Callback cli

### LINKED ISSUE #435: Add callback cli command
Create a callback associated with a promise.

Commands:
- create

Examples:
```
resonate callbacks create --promise-id foo --root-promise-id bar --timeout 1h --recv default
```

A receiver can be either a string or an object, an object must always have a type and data.
- logical as string (eg: "default")
- physical as url string (eg: "poll://default/6fa89b7e-4a56-40e8-ba4e-78864caa3278")
- physical as object (eg: `{"type": "poll", "data": {"group": "default", "id": "6fa89b7e-4a56-40e8-ba4e-78864caa3278"}}`

See [promises](https://github.com/resonatehq/resonate/tree/main/cmd/promises) and [schedules](https://github.com/resonatehq/resonate/tree/main/cmd/schedules) for a reference.

### PR DESCRIPTION
**Pull request by Madraceee**
### Description
Added callback cli. Invokes the inbuilt commands.

**This is not finished**

1. What should the value of requestID be. UUID or some string? 
2. For url and default, how should those values be passed?

Sry I was busy with my work, could not look into what should be passed for callbacks which dont require a json

**To-Do**
- [ ] Fix RequestID
- [ ] Fix non json flow

### CODE REVIEW

**Comment by Madraceee** [comment]
@dfarr I couldnt find an option to request for review , so can you help me out?

File: cmd/callbacks/create.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		recv          string
+	)
+	cmd := &cobra.Command{
+		Use:     "create ",
+		Short:   "Create callbacks",
+		Example: createCallbacksExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// What should requestID be?
+			paramsByte, err := json.Marshal(struct {
```
Comment (dfarr) [inline cmd/callbacks/create.go:39]: We can remove requestId from the CLI, its an optional parameter that is more handy for web applications, you can pass nil to CreateCallbackWithResponse for the params.

File: cmd/callbacks/create.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+					cmd.Printf("Callback exists for the promise: %s\n", promiseId)
+				} else {
+					cmd.PrintErrln(res.Status(), string(res.Body))
+				}
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&promiseId, "promise-id", "", "promise id")
```
Comment (dfarr) [inline cmd/callbacks/create.go:98]: We can mark this one as required

File: cmd/callbacks/create.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	cmd.Flags().StringVar(&rootPromiseId, "root-promise-id", "", "root promise id")
```
Comment (dfarr) [inline cmd/callbacks/create.go:99]: We can also mark this one as required

File: cmd/callbacks/create.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	cmd.Flags().DurationVar(&timeout, "timeout", 0, "callback timeout")
```
Comment (dfarr) [inline cmd/callbacks/create.go:100]: I think we can put a sensible default here, maybe 1m?

File: cmd/callbacks/create.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	cmd.Flags().StringVar(&recv, "recv", "", "receive object")
```
Comment (dfarr) [inline cmd/callbacks/create.go:101]: We can put default (literally) as the default for recv

**Review by dfarr — CHANGES_REQUESTED** [review]
So sorry for the delay in getting back to you on this PR. The `recv` flag is so tricky because it can be either a string or a json object in the form `{"type": "", data: {}}`.

I played around a little with this, what do you think about this approach?

```go
var recv v1.Recv

if json.Valid([]byte(recvStr)) {
	var recv0 v1.Recv0

	if err := json.Unmarshal([]byte(recvStr), &recv0); err != nil {
		return err
	}
	if err := recv.FromRecv0(recv0); err != nil {
		return err
	}
} else {
	if err := recv.FromRecv1(recvStr); err != nil {
		return err
	}
}
```

This unmarshals the provided value to a `v1.Recv0` if the value provided by the user is valid json, otherwise we assume it's a string. That way the user can provide one or the other, for example:

```
# string example
go run ./... callbacks create --promise-id foo --root-promise-id bar --recv default

# json example
go run ./... callbacks create --promise-id foo --root-promise-id bar --recv '{"type":"poll", "data":{"group":"default","id":"0"}}'
```

**Comment by Madraceee** [comment]
I am not well. I will look into this next week. Is that fine?

**Comment by dfarr** [comment]
> I am not well. I will look into this next week. Is that fine?

Of course, take your time and hope you get well soon!

**Comment by dfarr** [comment]
Hey @Madraceee I can take over this PR if you'd like? Let me know your status :)

**Comment by dfarr** [comment]
Closing in favour of #499

**Pull request closed without merging**